projectile.el

```diff
@@ -863,17 +863,20 @@ topmost sequence of matched directories.  Nil otherwise."
 (defun projectile-project-root ()
   "Retrieves the root directory of a project if available.
 The current directory is assumed to be the project's root otherwise."
-  (let ((dir default-directory))
-    (or (cl-some
-         (lambda (func)
-           (let* ((cache-key (format "%s-%s" func dir))
-                  (cache-value (gethash cache-key projectile-project-root-cache)))
-             (if (and cache-value (file-exists-p cache-value))
-                 cache-value
-               (let ((value (funcall func (file-truename dir))))
-                 (puthash cache-key value projectile-project-root-cache)
-                 value))))
-         projectile-project-root-files-functions)
+  (let* ((dir default-directory)
+         (is-local (not (file-remote-p dir)))
+         (is-connected (file-remote-p default-directory nil t)))
```

Comment:
huh? `is-local` is true if the file is local, `is-connected` is true if the file is remote but we are already connected.